// cpf-mask-ng2/bundles/amazing.umd.js
(function (global, factory) {
	typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core')) :
	typeof define === 'function' && define.amd ? define(['exports', '@angular/core'], factory) :
	(factory((global.ng = global.ng || {}, global.ng.amazing = global.ng.amazing || {}),global.ng.core));
}(this, (function (exports,_angular_core) { 'use strict';

var VMasker = require('vanilla-masker');
var CPFMask = (function () {
    function CPFMask(element, render) {
        this.element = element;
        this.render = render;
        console.log(element);
        this.nativeElement = this.element.nativeElement;
        VMasker(this.nativeElement).maskPattern('999.999.999-99');
    }
    CPFMask.decorators = [
        { type: _angular_core.Directive, args: [{
                    selector: '[CPFMask]'
                },] },
    ];
    /** @nocollapse */
    CPFMask.ctorParameters = function () { return [
        { type: _angular_core.ElementRef, },
        { type: _angular_core.Renderer, },
    ]; };
    return CPFMask;
}());

var CPFMaskModule = (function () {
    function CPFMaskModule() {
    }
    CPFMaskModule.decorators = [
        { type: _angular_core.NgModule, args: [{
                    declarations: [CPFMask],
                    imports: [],
                    exports: [CPFMask],
                    providers: [],
                },] },
    ];
    return CPFMaskModule;
}());

exports.CPFMaskModule = CPFMaskModule;
exports.CPFMask = CPFMask;

Object.defineProperty(exports, '__esModule', { value: true });

})));
